osu.Game/Graphics/Containers/BeatSyncedContainer.cs

```diff
@@ -42,25 +45,49 @@ protected override void Update()
             if (currentTrackTime < timingPoint.Time)
                 beatIndex--;
 
+            // We are at the lead-in part
+            if (beatIndex < 0)
+                return;
+
             if (timingPoint == lastTimingPoint && beatIndex == lastBeat)
                 return;
 
             double offsetFromBeat = (timingPoint.Time - currentTrackTime) % timingPoint.BeatLength;
 
+            // Beat can be trimmed by the next timing point, so in some cases we can't use timingPoint.BeatLength
+            double beatLength = calculateBeatLength(timingPoint, getNextTimingPoint(timingPoint), currentTrackTime);

```

Comment:
as i mentioned in discord, i don't believe we *want* to clip the beatlength in cases like this. it creates uneven effects.